packages/core/src/lib/base.ts

```diff
@@ -149,8 +155,10 @@ export abstract class FrontendBase<B extends Backend, O extends Options>
       ? beforeBreadcrumb(mergedBreadcrumb)
       : mergedBreadcrumb;
 
-    const breadcrumbs = await this.getBreadcrumbs();
-    this.breadcrumbs = [...breadcrumbs, finalBreadcrumb].slice(-maxBreadcrumbs);
+    await this.getBreadcrumbs();
```

Comment:
missed `const breadcrumbs = `, unless you just wanted to trigger the event loop. I'd change it nonetheless, as it's confusing.

Reply:
That would be a race condition ("lost update"). I agree we should fix the API somehow, but there is no way to fix this with the current asynchronous functions. The only correct way would be:

```js
await this.loadBreadcrumbs();
this.breadcrumbs = [...this.breadcrumbs, xyz];
```